src/GitHub.VisualStudio.UI/Base/TeamExplorerItemBase.cs

```diff
@@ -75,21 +74,51 @@ public virtual void Invalidate()
 
         void SubscribeToRepoChanges()
         {
-            holder.Subscribe(this, (ILocalRepositoryModel repo) =>
+            UpdateRepo(holder.TeamExplorerContext.ActiveRepository);
+            holder.TeamExplorerContext.PropertyChanged += TeamExplorerContext_PropertyChanged;
+            holder.TeamExplorerContext.StatusChanged += TeamExplorerContext_StatusChanged;
+        }
+
+        void TeamExplorerContext_StatusChanged(object sender, EventArgs e)
+        {
+            UpdateRepoOnMainThread(holder.TeamExplorerContext.ActiveRepository);
+        }
+
+        void TeamExplorerContext_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(holder.TeamExplorerContext.ActiveRepository))
             {
-                var changed = !Equals(ActiveRepo, repo);
-                ActiveRepo = repo;
-                RepoChanged(changed);
-            });
+                UpdateRepoOnMainThread(holder.TeamExplorerContext.ActiveRepository);
+            }
+        }
+
+        void UpdateRepoOnMainThread(ILocalRepositoryModel repo)
+        {
+            holder.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await holder.JoinableTaskFactory.SwitchToMainThreadAsync();
+                UpdateRepo(repo);
+            }).Task.Forget();
         }
 
         void Unsubscribe()
         {
-            holder.Unsubscribe(this);
+            //holder.Unsubscribe(this);
```

Comment:
Commented-out code.

Reply:
Fixed in #2028 here https://github.com/github/VisualStudio/pull/2028/commits/bf4c2f9026d69b23986ea2a12660cf96746d5180.